ruby_event_store-rom/lib/ruby_event_store/rom/repositories/events.rb

```diff
@@ -42,29 +42,15 @@ def last_stream_event(stream)
         end
 
         def read(specification)
-          unless specification.head?
-            offset_entry_id = stream_entries.by_stream_and_event_id(specification.stream, specification.start).fetch(:id)
-          end
-
-          direction = specification.forward? ? :forward : :backward
-
-          limit = specification.limit if specification.limit?
-
-          if specification.last? && specification.head?
-            direction = specification.forward? ? :backward : :forward
-          end
-
-          query = stream_entries.ordered(direction, specification.stream, offset_entry_id)
-
-          query = query.by_event_id(specification.with_ids) if specification.with_ids
-          query = query.by_event_type(specification.with_types) if specification.with_types?
+          query = read_scope(specification)
 
           if specification.batched?
             reader = ->(offset, limit) do
               query_builder(query, offset: offset, limit: limit).to_ary
             end
-            BatchEnumerator.new(specification.batch_size, limit || Float::INFINITY, reader).each
+            BatchEnumerator.new(specification.batch_size, specification.limit, reader).each
```

Comment:
Missed change from a different commit?

Reply:
Nope - just simplification.

Was:
```
limit = specification.limit if specification.limit?
and later as argument: limit || Float::INFINITY
```

but that's not needed because `specification.limit` is `Float::INFINITY` if limit has not been specified.